tests/test_as_path_verification.c

```diff
@@ -14,108 +14,92 @@
 #include <string.h>
 #include <assert.h>
 
-static struct aspa_record *create_aspa_record(uint32_t cas, uint32_t *provider_asns, size_t provider_count)
-{
-	struct aspa_record *aspa_record = lrtr_malloc(sizeof(struct aspa_record));
-	aspa_record->customer_asn = cas;
-
-	size_t size = provider_count * sizeof(uint32_t);
-	aspa_record->provider_asns = lrtr_malloc(size);
-	memcpy(aspa_record->provider_asns, provider_asns, size);
 
-	aspa_record->provider_count = provider_count;
+#define ASNS(...) (uint32_t []) { __VA_ARGS__ }
 
-	return aspa_record;
+#define RECORD(cas, providers) (struct aspa_record) { \
+	.customer_asn = cas, \
+	.provider_count = (size_t)(sizeof(providers) / sizeof(uint32_t)), \
+	.provider_asns = sizeof(providers) == 0 ? NULL : providers \
 }
 
-static enum aspa_status aspa_store_insert_node(struct aspa_store_node **store, struct rtr_socket *rtr_socket,
-					       struct aspa_array *aspa_array)
-{
-	// Allocate new node
-	struct aspa_store_node *new = lrtr_malloc(sizeof(struct aspa_store_node));
-
-	if (new == NULL)
-		return ASPA_ERROR;
-
-	// Store socket and ASPA array
-	new->rtr_socket = rtr_socket;
-	new->aspa_array = aspa_array;
-
-	// prepend new node
-	new->next = *store; // may be NULL
-	*store = new;
-
-	return ASPA_SUCCESS;
+#define ADD_OPERATION(idx, rec) ((struct aspa_update_operation) { \
+	.index = idx, \
+	.record = rec, \
+	.type = ASPA_ADD, \
+	.is_no_op = false \
+})
+
+#define BUILD_ASPA_TABLE(...) \
+	struct aspa_table *aspa_table = lrtr_malloc(sizeof(*aspa_table)); \
+	assert(aspa_table != NULL); \
+	aspa_table_init(aspa_table, NULL); \
+	\
+	ENTER_ASPA_TABLE_NEW_SOCKET(aspa_table, __VA_ARGS__) \
+
+
+#define _CAT_(a, b) a ## b
+#define _CAT(a, b) _CAT_(a, b)
+#define ENTER_ASPA_TABLE_NEW_SOCKET(aspa_table, ...) { \
```

Comment:
why do we need a separate macro? Can't this be inlined into `BUILD_ASPA_TABLE`? I think I'm missing something here...

Reply:
Yes, we need it split for `test_create_aspa_table()`

Reply:
alright, then I think this should be called `ADD_RECORDS`

Reply:
also, I think it's kind of troublesome that `BUILD_ASPA_TABLE` implicitly defines a table `VERIFY_AS...` then uses implicitly..

Reply:
I made table names explicit now, but for the BUILD_ASPA macro to be able to return something, it would need to be a function, causing more rewrites and scoping issues for `VERIFY_AS_PATH` / `test_create_aspa_table`

Reply:
great, ready to merge.